src/app/common/components/PipelineSummary.tsx

```diff
@@ -1,19 +1,101 @@
 import * as React from 'react';
+import { Flex, FlexItem, Text } from '@patternfly/react-core';
+import {
+  ResourcesAlmostEmptyIcon,
+  ResourcesAlmostFullIcon,
+  ResourcesFullIcon,
+} from '@patternfly/react-icons';
 
-import { Text } from '@patternfly/react-core';
+import {
+  global_danger_color_100 as dangerColor,
+  global_disabled_color_200 as disabledColor,
+  global_info_color_100 as infoColor,
+  global_success_color_100 as successColor,
+} from '@patternfly/react-tokens';
+
+import { IVMStatus } from '@app/queries/types';
+import './PipelineSummary.css';
+import { MigrationVMStepsType } from '@app/common/constants';
 
 interface IPipelineSummaryProps {
-  total: number;
-  current: number;
-  status: string;
+  status: IVMStatus;
 }
 
+const Dash = (isReached: boolean): JSX.Element => {
+  return (
+    <FlexItem alignSelf={{ default: 'alignSelfCenter' }}>
+      {isReached ? <div className="dash dashReached" /> : <div className="dash dashNotReached" />}
+    </FlexItem>
+  );
+};
+
 const PipelineSummary: React.FunctionComponent<IPipelineSummaryProps> = ({
-  total,
-  current,
   status,
 }: IPipelineSummaryProps) => {
-  return <Text>{status}</Text>;
+  let title: string;
+  let summary: JSX.Element;
+
+  const Chain = (Face, times, color) => {
```

Comment:
I was thinking instead of a plain function this should be a `React.FunctionComponent<IChainProps>` and Face, times, color could be moved to an `IChainProps` interface and given types. Then below instead of `        `{Chain(Face, times - 1, color)}` you'd use `<Chain Face={Face} times={times - 1} color={color} />`. Same for usages below for the `full` and `empty` variables.

The distinction is a bit pedantic, but these definitely follow the component model (a function of props that returns JSX) so I think they fit better in that syntax.

Reply:
Sure but what about the rendering?

Reply:
I'm not sure what you mean? You should just be able to embed the components `<LikeThis>` instead of `{likeThis()}` and it should render fine

Reply:
@mturley, oh I meant if there is extra rendering depending on the type (JSX vs React Elements).

Reply:
That's what I guess I'm not understanding, I don't see the real difference between `JSX.Element` and `React.ReactNode`. Apparently they are mostly interchangeable (`ReactNode` is [apparently a little more inclusive of things like arrays](https://github.com/typescript-cheatsheets/react/blob/cc2ec67e0b36b7e838df8f2a54b4cd84df6c78cd/README.md#useful-react-prop-type-examples)). So it seems like these should all just be components, and therefore return `React.ReactNode`.

What extra rendering are you referring to?

Reply:
Well the extra rendering I meant is because I don't clearly see the demarcation line between the two either.
So is it ReactNode rendering a JSX? in which case a JSX element wouldn't save that step, besides the whole result of the current component being rendered after.